File: 1732-find-the-highest-altitude/1732-find-the-highest-altitude.cpp
#include<vector>
#include<algorithm> // to use max function

class Solution {
public:
    int largestAltitude(vector<int>& gain) {
        int altitude = 0;
        int maxAltitude = 0;
        
        for (int i = 0; i < gain.size(); i++) {
            altitude += gain[i];
            maxAltitude = max(maxAltitude, altitude);
        }
        
        return maxAltitude;
    }
};

File: 1732-find-the-highest-altitude/1732-find-the-highest-altitude.py
class Solution:
    def largestAltitude(self, gain: List[int]) -> int:
        altitude = 0
        maxAltitude = 0
        length = len(gain)
        
        for i in range(0, length):
            altitude += gain[i]
            maxAltitude = max(maxAltitude, altitude)
        
        return maxAltitude

